### PR TITLE
Chore: Errors follow-up client clean up and refactor

### PR DESCRIPTION
**Pull request by aolfat**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: fferr/errors.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ type Error interface {
 	GRPCStatus() *status.Status
 	ToErr() error
 	AddDetail(key, value string)
+	AddDetails(keysAndValues ...interface{})
```
Comment (epps) [inline fferr/errors.go:79]: Do we really want to allow values of any type here? Why not limit it to strings to keep things simple?

Comment (aolfat) [reply fferr/errors.go:79]: It's just the interface method to do it. I modeled this after how logger.Infow works. The validation is done at the impl level

File: fferr/generic.go [hunk 2]
```diff
@@ -44,6 +44,32 @@ func (e *GenericError) AddDetail(key, value string) {
...
+	}
+
+	for i := 0; i < n; i += 2 {
+		key, ok := keysAndValues[i].(string)
+		if !ok {
+			// Handle or log the error if the key is not a string.
+			fmt.Printf("Key at index %d is not a string\n", i)
+			continue
+		}
+		// Convert the value to a string, this assumes that the value has a meaningful string representation
```
Comment (epps) [inline fferr/generic.go:67]: I don't think we should be making this assumption. I think we should force the consumer to use a string value so it's clear to them that without a meaningful string representation, they can't add details.

**Review by epps — APPROVED** [review]
👍

**Review by ahmadnazeri — APPROVED** [review]
LGTM, agree with Erik that we should expect it to be strings

**Pull request merged**